### DISCUSSION TITLE
Are lumped elements series or parallel?

### DISCUSSION THREAD

**Discussion by AndrewKrill** [thliebig/openEMS-Project#240] [Q&A Usage]
The documentation really isn't clear- at least what I've been able to find.

I assumed they would be serial, but after my sim came out different than I expected, and seeing an example with C=10p and R=10000, I'm not so sure. Really don't want to waste the time trying both cases. Hoping someone else happens to know.

Thanks

**Comment by VolkerMuehlhaus** [comment]
If you look at the C++ code in line 125 and below, you find this:

`// The default is a parallel lumped circuit, in case none is supplied
`
https://github.com/thliebig/CSXCAD/blob/master/src/CSPropLumpedElement.cpp

**Reply by AndrewKrill** [reply]
So just to confirm... If I'm using matlab interface...

CSX = AddLumpedElement( CSX, 'HorizL', 0, 'Caps', 1, 'L', 47e-9, 'R', 0.003, 'LEtype', 1);

Should be a series RL?

**Reply by VolkerMuehlhaus** [reply]
@AndrewKrill Sorry Andrew, I don't know how to use those optional parameteres in the Matlab interface.

**Comment by thliebig** [comment]
The feature to switch the type LEtype is pretty new I think and I believe it is not even in the currently released version.
Thus I think this is still a simple lossy capacitor, meaning it is "parallel"...
I hope this helps.

**Reply by AndrewKrill** [reply]
What about inductors?

```
CSX = AddLumpedElement( CSX, 'HorizL', 0, 'Caps', 1, 'L', 47e-9);

Operator::Calc_LumpedElements(): Warning: Lumped Element R or C not specified! skipping.  ID: 50 @ Property: HorizL
```

**Reply by AndrewKrill** [reply]
I'm confused, looking at the code...

Are lumped inductors not implemented? Or do they require a resistance term?

**Reply by thliebig** [reply]
I think with the current release version only capacitors, resistors or lossy capacitors (C+R in parallel) are possible.

**Reply by gadiLhv** [reply]
You updated that PR, if I recall correctly, @thliebig. It should support inductors as well.

@AndrewKrill, 
[Inductor support in parallel](https://github.com/thliebig/openEMS/blob/7d7688a288bdbd16adc5e5d9cf9ac51ae75a5be7/FDTD/extensions/operator_ext_lumpedRLC.cpp#L313)

[Inductor support in series](https://github.com/thliebig/openEMS/blob/7d7688a288bdbd16adc5e5d9cf9ac51ae75a5be7/FDTD/extensions/operator_ext_lumpedRLC.cpp#L265C1-L273C59)

In series it's more implicit, you need to go to the [PR](https://github.com/thliebig/openEMS/pull/121) to understand it better. The [mathematical formulation](https://github.com/thliebig/openEMS/files/12111853/Lumped_RLC_Formulation.pdf) is also there.

**Comment by gadiLhv** [comment]
This is actually related to [CSXCAD](https://github.com/thliebig/CSXCAD/tree/9257bf17b6bbce85ea00d9f85b89416530563cd2) rather than OpenEMS. 

The relevant code snippet is [here](https://github.com/thliebig/CSXCAD/blob/9257bf17b6bbce85ea00d9f85b89416530563cd2/src/CSPropLumpedElement.h#L29C1-L34C4):
```cpp

	enum LEtype
	{
		PARALLEL = 0,
		SERIES = 1,
		INVALID = -1
	};

```
